app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala

```diff
@@ -107,6 +112,75 @@ object NSIController extends BaseController with Logging {
 
   private val ninoStatusRegex = """AS(\d{3}).*""".r
 
+  def getAccount(correlationId: Option[String], nino: Option[String], version: Option[String], systemId: Option[String]): Action[AnyContent] = Action { implicit request ⇒
+    validateParams(correlationId, nino, version, systemId)
+      .fold(
+        errors ⇒ {
+          logger.warn("[Handle Account Query] invalid params")
+          errors match {
+            case error :: Nil ⇒ BadRequest(Json.toJson(error))
+            case _            ⇒ BadRequest
+          }
+        }, {
+          case nino ⇒
+            if (nino.contains("401")) {
+              Unauthorized
+            } else if (nino.contains("500")) {
+              InternalServerError
+            } else {
+              val maybeAccount = getAccountByNino(nino, correlationId)
+              maybeAccount match {
+                case Right(a) ⇒ Ok(Json.toJson(a))
+                case Left(e)  ⇒ BadRequest(Json.toJson(e))
+              }
+            }
+          case other ⇒
+            BadRequest
+        })
+  }
+
+  def getTransactions: Action[AnyContent] = ???
+
+  def getMessages: Action[AnyContent] = ???
+
+  def getMessage(messageId: String): Action[AnyContent] = ???
+
+  private def validateParams(correlationId: Option[String], nino: Option[String], version: Option[String], systemId: Option[String]): Either[List[NSIErrorResponse], NINO] = {
+
+    val versionValidation: ValidatedNel[NSIErrorResponse, String] = version.fold[Validated[NSIErrorResponse, String]](
+      Validated.Invalid(NSIErrorResponse.missingVersionResponse(correlationId))
+    )(
+        v ⇒ if (v =!= "V1.0") Validated.Invalid(NSIErrorResponse.unsupportedVersionResponse(correlationId)) else Validated.Valid(v)
+      ).toValidatedNel
+
+    //    val systemIdValidation: ValidatedNel[NSIErrorResponse, List[String]] = map.get("systemId").fold[Validated[NSIErrorResponse, List[String]]](
```

Comment:
I'd usually not leave commented out code in when merging to master - it can be retrieved from git history if necessary (actually, maybe not if we squash the commits, but this is very similar to the validation for other params so should be easy to redo if needed).

Can leave it in if you like as long as you remember to remove it when we have an answer from Rob if it isn't needed.

Reply:
ok I have removed it.